WordPress/gutenberg
# Prevent network requests related to ephemeral posts in the widgets editor

## Description

This PR follows up the discussion from https://github.com/WordPress/gutenberg/pull/25831:

> Both navigation editor and widgets editor need a top-level entity record to store all the blocks. While post editor may just use the post entity that comes from the API, there is no corresponding concept in the navigation editor or the widgets editors. At the moment both editors use a "fake" post entity record that's never resolved from the API or persisted to the API. This requires carefully dispatching actions to avoid triggering any resolvers. In other words, it's a hack.

The approach taken here is different. Instead of adding a new concept to `core-data`, this PR simply plugs into `api-fetch` and prevents any unwanted requests from happening. While the implementation differs, the approach is inspired by [Drupal's api-fetch.](https://git.drupalcode.org/project/gutenberg/-/blob/8.x-2.x/js/api-fetch.js)

The advantage of this approach over the current approach is that it's resilient. It doesn't require any initialization, it doesn't rely on implementation details of the resolution pipeline (such as faking `startResolution` and `endResolution`), it won't be easily broken by updates to `core-data`. The only hardcoded part is the path string, I'd rather not use the one specified in `defaultEntities -> baseURL` as the data structure there is just another implementation detail of the resolution system.

The same treatment could be applied to the navigation editor which also uses an in-memory post as a convenience wrapper.

## How has this been tested?
1. Go to widgets editor.
1. Open the network panel and watch out for any 404 and 403.
1. Confirm it works as before this PR - widgets are loaded, possible to manipulate, saving works fine.
1. Confirm there weren't any `widget-area` related network error logged.

## Types of changes
Non-breaking change

## Checklist:
